networkx/algorithms/dag.py

```diff
@@ -415,16 +420,16 @@ def dag_longest_path(G):
     dag_longest_path_length
     """
     dist = {}  # stores [node, distance] pair
-    for node in nx.topological_sort(G):
+    for v in nx.topological_sort(G):
         # pairs of dist,node for all incoming edges
-        pairs = [(dist[v][0] + 1, v) for v in G.pred[node]]
+        pairs = [(dist[u][0] + G[u][v].get(weight, default_weight), u) for u in G.pred[v]]
```

Comment:
Saves lookup to use ```.items()```

    pairs = [(dist[u][0] + data.get(weight, default_weight), u) for u,data in G.pred[v].items()]

We should consider whether this will be much slower for the unweighted case.
If so, could have ```weight=None``` in the function arguments and if ```None``` then we do old loop. Code gets bigger, but not by much. Depends on how much slowdown there is.